### PR TITLE
[Custom Fields] Show an entry point in product details screen

### LINKED ISSUE #12677: [Custom Fields] Add an entry point from the product details

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #12677 

### Description
This PR adds logic to show an entry point to the custom fields screen from the product details, the entry point will be shown either as a row if the product had already some custom fields, or in the details bottom sheet when it doesn't.

**Note:** We don't have an empty view for the custom fields screen yet, it will be added later.

### Steps to reproduce
1. Open the app.
2. Open a product details.

### Testing information
- Confirm that the custom fields row is shown when the product has already some fields.
- Confirm that the custom fields button is shown in the details bottom sheet when no custom fields were added yet.
- Test the navigation logic for both cases.
- Make changes to the custom fields and confirm the logic is reflected.

### Images/gif
| Product with existing custom fields | Product with no custom fields |
| --- | --- |
| Screenshot_20240925_092727 | Screenshot_20240925_092737 |

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/238">20.6</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -2007,6 +2007,8 @@
     <string name="product_download_limit_count">%d downloads</string>
     <string name="product_download_expiry">Expiry</string>
     <string name="product_download_expiry_days">%d days</string>
+    <string name="product_custom_fields">Custom Fields</string>
+    <string name="product_custom_fields_desc">View and edit Custom Fields</string>
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/res/values/strings.xml]: Minor np, is there a reason to capitalize the `Custom Fields` part of this description?

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/products/details/ProductDetailBottomSheetBuilderTest.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class ProductDetailBottomSheetBuilderTest : BaseUnitTest() {
```
Comment (JorgeMucientes) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/products/details/ProductDetailBottomSheetBuilderTest.kt:19]: 👍🏼 🥇

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/products/details/ProductDetailCardBuilderTest.kt [hunk 3]
```diff
@@ -138,4 +150,47 @@ class ProductDetailCardBuilderTest : BaseUnitTest() {
 
         Assert.assertTrue("Expected a Product card with Quantity Rules", foundQuantityRulesCard)
     }
+
+    @Test
+    fun `when a product has no displayable fields, then hide the custom fields card`() = testBlocking {
```
Comment (JorgeMucientes) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/products/details/ProductDetailCardBuilderTest.kt]: Maybe you'll find it to much, but these first couple tests could benefit from adding something like `given product is saved on server, when...` ? 
Wdyt?

**Review by JorgeMucientes — APPROVED** [review]
Excellent job @hichamboushaba. Works great. I added a couple super minor np. Feel free to ignore them.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/products/details/ProductDetailCardBuilderTest.kt (hunk 3 above)
Comment (hichamboushaba) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/products/details/ProductDetailCardBuilderTest.kt]: Good suggestion, applied.

File: WooCommerce/src/main/res/values/strings.xml (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/res/values/strings.xml]: Changed in the last commits, thanks.

**Pull request merged**